internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2211,6 +2220,13 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 			}
 		}
 	}
+
+	if key := "node_provisioning_mode"; d.HasChange(key) {
+		updateCluster = true
+		nodeProvisioningMode := d.Get(key).(string)
+		existing.Model.Properties.NodeProvisioningProfile.Mode = pointer.To(managedclusters.NodeProvisioningProfileMode(nodeProvisioningMode))
+	}
```

Comment:
I think the AKS API doesn't support provisioning mode changing after cluster creation so this probably isn't quite right

Reply:
@Chippiewill Based on the public API documentation, it IS possible to change node provisioning mode:

_"The set of default Karpenter NodePools (CRDs) configured for node provisioning. This field has no effect unless mode is 'Auto'. Warning: Changing this from Auto to None on an existing cluster will cause the default Karpenter NodePools to be deleted, which will drain and delete the nodes associated with those pools. It is strongly recommended to not do this unless there are idle nodes ready to take the pods evicted by that action. If not specified, the default is Auto. For more information see aka.ms/aks/nap#node-pools."_

https://learn.microsoft.com/en-us/rest/api/aks/managed-clusters/create-or-update?view=rest-aks-2025-05-01&tabs=HTTP#managedclusternodeprovisioningprofile